partner_preview_scripts/ot_crispr.py

```diff
@@ -25,142 +34,166 @@ def __init__(self, study_url: str) -> None:
+        gene_id = gene_id.split("_")[0]
 
         return gene_id
 
     @staticmethod
     def parse_MAGeCK_file(row: pd.Series) -> pd.DataFrame:
         """This function returns a pandas dataframe with the datafile and with properly named columns"""
 
-        datafile = row['dataFile']
-        filterColumn = row['filterColumn']
-        threshold = float(row['threshold'])
-        studyId = row['studyId']
-        controlDataFile = row['ControlDataset']
+        datafile = row["dataFile"]
+        filterColumn = row["filterColumn"]
+        threshold = float(row["threshold"])
+        studyId = row["studyId"]
+        controlDataFile = row["ControlDataset"]
 
         # Read data, filter and rename columns:
         mageck_df = (
-            pd.read_csv(datafile, sep='\t')
-            .rename(columns={
-                filterColumn: 'resourceScore',
-                'id': 'targetFromSourceId',
-                'neg|lfc': 'log2FoldChangeValue'
-            })
-            .loc[lambda df: df.resourceScore <= threshold]
-            [['targetFromSourceId', 'resourceScore', 'log2FoldChangeValue']]
+            pd.read_csv(datafile, sep="\t")
+            .rename(
+                columns={
+                    filterColumn: "resourceScore",
+                    "id": "targetFromSourceId",
+                    "neg|lfc": "log2FoldChangeValue",
+                }
+            )
+            .loc[lambda df: df.resourceScore <= threshold][
+                ["targetFromSourceId", "resourceScore", "log2FoldChangeValue"]
+            ]
             .assign(studyId=studyId)
         )
 
         # Applying control if present:
         if pd.isna(controlDataFile):
-            logging.info(f'Number of genes reach threshold: {len(mageck_df)}')
+            logging.info(f"Number of genes reach threshold: {len(mageck_df)}")
             return mageck_df
 
         # Read control data, filter and rename columns:
-        logging.info(f'Reading control data file: {controlDataFile}')
+        logging.info(f"Reading control data file: {controlDataFile}")
         controlHits = (
-            pd.read_csv(controlDataFile, sep='\t')
-            .rename(columns={filterColumn: 'resourceScore', 'id': 'targetFromSourceId'})
-            .loc[lambda df: df.resourceScore <= threshold]
-            ['targetFromSourceId']
+            pd.read_csv(controlDataFile, sep="\t")
+            .rename(columns={filterColumn: "resourceScore", "id": "targetFromSourceId"})
+            .loc[lambda df: df.resourceScore <= threshold]["targetFromSourceId"]
             .tolist()
         )
 
         # Excluding control genes:
         mageck_df = mageck_df.loc[lambda df: df.targetFromSourceId.isin(controlHits)]
 
-        logging.info(f'Number of genes reach threshold: {len(mageck_df)}')
+        logging.info(f"Number of genes reach threshold: {len(mageck_df)}")
         return mageck_df
 
     def write_evidence(self, output_file: str) -> None:
```

Comment:
Why not `pd.to_json()`?

Reply:
Mostly because of this: `row.dropna()`: this statement allows dropping keys with null values.